### PR TITLE
Before and After Filter in Segments | Fetch from TypeOperatorProvider

### PR DESCRIPTION
**Pull request by avikarshasaha**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Steps to test this PR:

1. Open this PR for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.

### CODE REVIEW

**Review by rahuld-dev — DISMISSED** [review]
Thanks @avikarshasaha

**Review by dadarya0 — DISMISSED** [review]
Thanks @avikarshasaha working fine , tested some fields.

**Review by rahuld-dev — APPROVED** [review]
re-approving after fix conflicts

File: Tests/Unit/EventListener/SegmentFiltersChoicesGenerateSubscriberTest.php [hunk 1]
```diff
@@ -324,6 +351,164 @@ public function testOnGenerateSegmentFilters(): void
...
+                                OperatorOptions::LESS_THAN_OR_EQUAL,
+                            ],
+                        ]);
+
+                        return true;
+                    }),
+                ],
+                [
+                    LeadEvents::OVERRIDE_OPERATOR_LABEL_FOR_FIELD_TYPE, // @phpstan-ignore-line
+                    $this->callback(function (OverrideOperatorLabelEvent $event) { // @phpstan-ignore-line
```
Comment (rahuld-dev) [inline Tests/Unit/EventListener/SegmentFiltersChoicesGenerateSubscriberTest.php]: We can remove `// @phpstan-ignore-line` as file is excluded from phpstan.

**Review by rahuld-dev — CHANGES_REQUESTED** [review]

**Review by rahuld-dev — APPROVED** [review]
Thanks @avikarshasaha

**Review by rohitp19 — APPROVED** [review]
Thanks @avikarshasaha 
Please remove Tests/Unit/EventListener/SegmentFiltersChoicesGenerateSubscriberTest.php from ignore list when Community PR with constants gets merged

**Pull request merged**